src/js/components/utils/useFetchTasksByProject.tsx

```diff
@@ -0,0 +1,12 @@
+import apiFetch from '~js/utils/api';
+import { addUrlParams } from '~js/utils/api';
+
+export default (
+  project: any
+) => {
+  let tasks;
+  apiFetch({ url: addUrlParams(window.api_urls.task_list(), { project: project.id }), dispatch: null }).then(data => {
+    tasks = data;
+  })
+  return tasks;
+};
```

Comment:
So we've got a race condition here, where this API call is triggered every time the component is re-rendered (which is not ideal), but then because this is not an `async` fn, `tasks` is returned as `undefined` _before_ the data returns from the API. There are a number of ways we could solve it.

If we _just_ want to solve the race condition, we could probably fix it by converting this to async:

```suggestion
export default (projectId?: string) => {
  const dispatch = useDispatch<ThunkDispatch>();
  const [tasks, setTasks] = useState<Task[]>();

  useEffect(() => {
    const fetchTasks = async () => {
      if (projectId && !tasks) {
        // Fetch tasks from API
        const response: Task[] | null = await apiFetch({
          url: addUrlParams(window.api_urls.task_list(), {
            project: projectId,
          }),
          dispatch,
        });
        setTasks(response || []);
      }
    };

    fetchTasks();
  }, [dispatch, projectId, tasks]);

  return tasks;
};
```